### DISCUSSION TITLE
timeline to drop python3.6 in lambda layers?

### DISCUSSION THREAD

**Discussion by deric4** [aws-powertools/powertools-lambda-python#1312] [Q&A] [answered]
Just a follow up on @michaelbrewer s issue: https://github.com/awslabs/aws-lambda-powertools-python/issues/1132

I noticed that the layers produced since `aws-lambda-powertools-python v1.26.1` include a botocore version (>=1.27.0) that [has since dropped support for python3.6](https://github.com/boto/botocore/blob/1.27.0/CHANGELOG.rst).

Doesn't look like there's actually been any breaking API changes, but seems.... wrong 🙃.

```console
[cloudshell-user@ip-10-0-38-189 ~]$ aws lambda get-layer-version-by-arn --arn "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:20"
{
    "Content": {
        "Location": "https://prod-04-2014-layers.s3.us-east-1.amazonaws.com/snapshots/017000801446/AWSLambdaPowertoolsPython-.......",
        "CodeSize": 10639105
    },
    "LayerArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython",
    "LayerVersionArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:20",
    "Description": "AWS Lambda Powertools for python layer, version 1.26.1",
    "CreatedDate": "2022-06-07T16:17:19.106+0000",
    "Version": 20,
    "CompatibleRuntimes": [
        "python3.6",
        "python3.7",
        "python3.8",
        "python3.9"
    ],
    "LicenseInfo": "MIT-0 License"
}


[cloudshell-user@ip-10-0-38-189 ~]$ aws lambda get-layer-version-by-arn --arn "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:21"
{
    "Content": {
        "Location": "https://prod-04-2014-layers.s3.us-east-1.amazonaws.com/snapshots/017000801446/AWSLambdaPowertoolsPython-....",
        "CodeSha256": "LBvy0dvMf6GrnxK2by6GQboe9ZFaKnlfSBrBwzizdI4=",
        "CodeSize": 10666965
    },
    "LayerArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython",
    "LayerVersionArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:21",
    "Description": "AWS Lambda Powertools for python layer, version 1.26.2",
    "CreatedDate": "2022-06-16T15:36:08.705+0000",
    "Version": 21,
    "CompatibleRuntimes": [
        "python3.6",
        "python3.7",
        "python3.8",
        "python3.9"
    ],
    "LicenseInfo": "MIT-0 License"
}

[cloudshell-user@ip-10-0-38-189 ~]$  aws lambda get-layer-version-by-arn --arn "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:22"
{
    "Content": {
        "Location": "https://prod-04-2014-layers.s3.us-east-1.amazonaws.com/snapshots/017000801446/AWSLambdaPowertoolsPython-..........",
        "CodeSha256": "Es4tYNQRJXZEca6RDz0NLFo/ZjFGYN0lo6w8DnzxDPI=",
        "CodeSize": 10211577
    },
    "LayerArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython",
    "LayerVersionArn": "arn:aws:lambda:us-east-1:017000801446:layer:AWSLambdaPowertoolsPython:22",
    "Description": "AWS Lambda Powertools for python layer, version 1.26.3",
    "CreatedDate": "2022-07-04T14:02:37.248+0000",
    "Version": 22,
    "CompatibleRuntimes": [
        "python3.6",
        "python3.7",
        "python3.8",
        "python3.9"
    ],
    "LicenseInfo": "MIT-0 License"
}
```

related:
https://github.com/awslabs/aws-lambda-powertools-python/issues/1147

https://github.com/awslabs/aws-lambda-powertools-python/pull/1148#discussion_r855572486

**Comment by heitorlessa** [comment, marked as answer]
Hey @deric4 thanks a lot for bringing this up - you can watch progress here: https://github.com/orgs/awslabs/projects/51/views/11?query=is%3Aopen+sort%3Aupdated-desc

**TL;DR**: We're going to drop next month as per [Lambda deprecation runtime policy](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtime-support-policy) and cut a major version (v2) to signal 3.6 is dropped.

@am29d is currently refactoring our Lambda Layer release pipeline we've just migrated to GitHub Actions so we can both expand to additional regions, and more importantly make customizations to Layers (it's quite rigid at the moment hence this issue). What's not decided yet is whether we should create a new Lambda Layer (`AWSLambdaPowertoolsPythonV2`) as part of the new major version next month.

What's your take in it?

**Reply by deric4** [reply]
super late, but just wanted to say thanks!
